packages/block-library/src/navigation/edit.js

```diff
@@ -158,12 +164,18 @@ function Navigation( {
 			? undefined
 			: false;
 
+	const directInsertValue = useMemo(
+		() => ( onlyLinkInnerBlocks ? DIRECT_INSERT : [] ),
+		[ onlyLinkInnerBlocks ]
+	);
+
```

Comment:
This is memoized to stop the directInsertValue array from getting recreated on every render, and causing an infinite loop because it changes innerBlocks, which in turn the Navigation function depends on to calculate the directInsertValue 😖 

I can't think of a better way to do it though. I think this logic should reside in the block, because it might vary a lot - Submenus are set to always use it (at least until we sort out mega menus), and conceivably there could be blocks using it depending on their nesting status or some other condition.